metagraph/src/graph/alignment/aligner_helper.hpp

```diff
@@ -412,6 +561,157 @@ std::ostream& operator<<(std::ostream& out, const Alignment<NodeType> &alignment
     return out;
 }
 
+template <typename NodeType>
+class AlignmentSuffix {
+  public:
+    AlignmentSuffix(const Alignment<NodeType> &alignment, const DBGAlignerConfig &config,
+                    size_t k)
+          : alignment_(&alignment),
+            config_(&config),
+            begin_it_(alignment_->get_query().data()),
+            end_it_(alignment_->get_query_end()),
+            ref_begin_it_(alignment_->get_sequence().data()),
+            ref_end_it_(ref_begin_it_ + alignment_->get_sequence().size()),
+            cigar_begin_(alignment_->get_cigar(), alignment_->get_clipping()),
+            cigar_end_(alignment_->get_cigar(),
+                       alignment_->get_cigar().end()
+                          - static_cast<bool>(alignment_->get_end_clipping())),
+            cigar_it_(cigar_begin_),
+            score_(alignment_->get_score()),
+            trim_(0),
+            k_(k) {
+        assert(cigar_begin_ <= cigar_end_);
+        assert(cigar_it_ <= cigar_end_);
+    }
+
+    AlignmentSuffix& operator++();
+    AlignmentSuffix& operator--();
+
+    auto get_score() const { return score_; }
+    std::string_view get_query() const {
+        return std::string_view(begin_it_, end_it_ - begin_it_);
+    }
+
+    std::string_view get_sequence() const {
+        return std::string_view(ref_begin_it_, ref_end_it_ - ref_begin_it_);
+    }
+
+    Cigar::Operator get_front_op() const { return *cigar_it_; }
+    const CigarOpIterator& get_op_it() const {
+        assert(cigar_begin_ <= cigar_it_);
+        assert(cigar_it_ <= cigar_end_);
+        return cigar_it_;
+    }
+
+    bool eof() const { return cigar_it_ == cigar_end_; }
+    bool reof() const { return cigar_it_ == cigar_begin_; }
+
+    const Alignment<NodeType>& data() const { return *alignment_; }
+
+    size_t get_trim() const { return trim_; }
+    size_t get_k() const { return k_; }
+
+  private:
+    typedef typename Alignment<NodeType>::score_t score_t;
+    const Alignment<NodeType> *alignment_;
+    const DBGAlignerConfig *config_;
+    const char *begin_it_;
+    const char *end_it_;
+    const char *ref_begin_it_;
+    const char *ref_end_it_;
+    CigarOpIterator cigar_begin_;
+    CigarOpIterator cigar_end_;
+    CigarOpIterator cigar_it_;
+    score_t score_;
+    size_t trim_;
+    size_t k_;
+};
+
+template <typename NodeType>
+std::ostream& operator<<(std::ostream &out, const AlignmentSuffix<NodeType> &suffix) {
+    out << Alignment<NodeType>(suffix);
+    return out;
+}
+
+template <typename NodeType>
+class AlignmentPrefix {
+  public:
+    AlignmentPrefix(const Alignment<NodeType> &alignment, const DBGAlignerConfig &config,
+                    const DeBruijnGraph &graph)
```

Comment:
Easier to read if they are all in one list (vertical or horizontal).

When we call a function, we can neglect this rule for parameters, but declarations are read more often, so it's worth trading a bit of space to improve the readability.

```suggestion
    AlignmentPrefix(const Alignment<NodeType> &alignment,
                    const DBGAlignerConfig &config,
                    const DeBruijnGraph &graph)
```